base/set.jl

```diff
@@ -191,6 +191,69 @@ function unique(f::Callable, C)
 end
 
 """
+    unique!(A::AbstractVector)
+
+Remove duplicate items as determined by [`isequal`](@ref), then return the modified `A`.
+
+```jldoctest
+julia> unique!([1, 1, 1])
+1-element Array{Int64,1}:
+ 1
+
+julia> A = [7, 3, 2, 3, 7, 5];
+
+julia> unique!(A);
+
+julia> A
+4-element Array{Int64,1}:
+ 7
+ 3
+ 2
+ 5
+```
+"""
+function unique!(A::AbstractVector)
+    try
+        Aissorted = issorted(A) || issorted(A, rev=true)
+    catch
+        Aissorted = false
+    end
+    if isempty(A)
+        return A
+    elseif Aissorted
+        # If A is sorted, then we only need to keep track of one element and add that to A
+        # every time that we see a new element.
+        idxs = eachindex(A)
+        y = first(A)
+        j = start(idxs)
+        for i in idxs
+            x = A[i]
+            if !isequal(x, y)
+                _, j = next(idxs, j)
```

Comment:
This is still incorrect: you need `j, state = next(idxs, state)`, and changing `j = start(idxs)` to `state = start(idxs)`.